fix #167: [FR] Redirect conjuged verbs to their infinitive form

lasconic opened the pull request:
- [x] Fixes #167 
- [x] Tests added/updated.
- [x] `./check.sh` run and eventual issues fixed.
- [x] Updated CONTRIBUTORS.md, if not already done in the past.

BoboTiG commented on scripts/convert.py:
```diff
@@ -53,6 +54,16 @@ def make_groups(words: Words) -> Groups:
     return groups
 
 
+def make_variants(words: Words) -> Variants:
```
Could you add a docstring please?

BoboTiG commented on scripts/get.py:
```diff
@@ -159,6 +159,10 @@ def find_section_definitions(
     definitions: List[Definitions] = []
 
+    # do not look for definions in french verb form section
+    if locale == "fr" and section.title.strip() == "{{S|verbe|fr|flexion}}":
```
For multiple flexions, we could do it that way:
```suggestion
    if locale == "fr" and section.title.strip().startswith("{{S|verbe|fr|flexion"):
```

BoboTiG commented on scripts/get.py:
```diff
@@ -395,7 +399,17 @@ def parse_word(word: str, code: str, locale: str, force: bool = False) -> Word:
+    variant = ""
+    if locale == "fr" and not definitions:  # Pure verb form, no definition
+        for title, s in parsed_sections.items():
+            if "{{S|verbe|fr|flexion}}" == title:
```
```suggestion
            if title.startswith("{{S|verbe|fr|flexion"):
```

BoboTiG commented on scripts/lang/fr/__init__.py:
```diff
@@ -106,6 +106,7 @@
     "{{S|verbe|fr|loc",
     "{{S|verbe|fr|num",
     "{{S|verbe|fr}",
+    "{{S|verbe|fr|flexion}}",
```
```suggestion
    "{{S|verbe|fr|flexion",
```

BoboTiG commented on scripts/lang/fr/__init__.py:
```diff
@@ -608,6 +609,8 @@
     # {{WSP|Panthera leo}}
     "WSP": "term(parts[1])",
+    # {fr-verbe-flexion|colliger|ind.i.3s=oui}}
+    "fr-verbe-flexion": "parts[1] if parts[1] != 'grp=3' else parts[2]",
```
Could you keep the dict sorted?

BoboTiG commented on scripts/stubs.py:
```diff
@@ -4,5 +4,5 @@
 Definitions = Union[str, Tuple[str, ...]]
 Parts = Tuple[str, ...]
-Word = namedtuple("Word", "pronunciation, genre, etymology, definitions")
+Word = namedtuple("Word", "pronunciation, genre, etymology, definitions, variant")
```
As there may be several variants, should we use plural here? If so, apply changes everywhere :)

lasconic replied:
There might be several variant indeed ("mirent" -> mirer / mettre) but in the end we need to choose one to add the word in the HTML file... So not sure it's worth making it plural. It's a string.

BoboTiG replied:
That's my point: variants could be a list of strings and we would have to handle "mirer" and "mettre". Maybe useless for now :)

BoboTiG commented on tests/test_fr.py:
```diff
@@ -380,6 +427,8 @@ def test_parse_word(word, pronunciation, genre, etymology, definitions, page):
             "<i>(XI<sup>e</sup> siècle)</i> Du moyen français <i>chambre</i>, de l’ancien français <i>chambre</i>, <i>cambre</i>, <i>cambra</i>, du bas latin <i>camera</i> (« pièce, chambre »), du latin classique <i>camera</i>, <i>camara</i> (« voute, plafond vouté »), du grec ancien καμάρα, <i>kamárā</i> (« voute, lieu couvert par une construction »).",  # noqa
         ),
+        ("{{fr-verbe-flexion|colliger|ind.i.3s=oui}}", "colliger"),
+        ("{{fr-verbe-flexion|grp=3|couvrir|ind.i.3s=oui}}", "couvrir"),
```
Could you keep the list sorted here too?

BoboTiG commented on scripts/get.py:
```diff
@@ -395,7 +399,17 @@ def parse_word(word: str, code: str, locale: str, force: bool = False) -> Word:
         nature = find_genre(code, genre[locale])
 
-    return Word(pron, nature, etymology, definitions)
+    # find if  variant and delete unwanted definitions
```
```suggestion
    # find if variant and delete unwanted definitions
```

BoboTiG commented on scripts/get.py:
```diff
@@ -159,6 +159,10 @@ def find_section_definitions(
     """Find definitions from the given *section*, with eventual sub-definitions."""
     definitions: List[Definitions] = []
 
+    # do not look for definions in french verb form section
```
```suggestion
    # do not look for definitions in french verb form section
```

lasconic commented:
I addressed most of the remarks. I'm not sure about "variant"/"variants". As I said, it's a single variant, we just need to choose. I chose the first one and broke the loop: https://github.com/BoboTiG/ebook-reader-dict/pull/191/files#diff-d00330496418af6b203d88de9efa2fffe42ccc66921d336b7455b852e77c0ca8R411

BoboTiG commented:
Great!
I merged to have the new words in the next auto-update. Let's handle next fixes/improvements in specific PRs.

Thanks for the work :)
